src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexEmbeddedBraceMatcher.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+{
+    using RegexToken = EmbeddedSyntaxToken<RegexKind>;
+
+    /// <summary>
+    /// Brace matching impl for embedded regex strings.
+    /// </summary>
+    internal class RegexEmbeddedBraceMatcher : IEmbeddedBraceMatcher
+    {
+        private readonly RegexEmbeddedLanguage _language;
+
+        public RegexEmbeddedBraceMatcher(RegexEmbeddedLanguage language)
+        {
+            _language = language;
+        }
+
+        public async Task<EmbeddedBraceMatchingResult?> FindBracesAsync(
+            Document document, int position, CancellationToken cancellationToken)
+        {
+            var option = document.Project.Solution.Workspace.Options.GetOption(
+                RegularExpressionsOptions.HighlightRelatedRegexComponentsUnderCursor, document.Project.Language);
+            if (!option)
+            {
+                return default;
```

Comment:
>default [](start = 23, length = 7)

nit: consider `null` (for consistency with other returns)

Reply:
indeed!